src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -387,13 +387,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $lowercase = true)
                         $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
                     }
 
-                    if ($strict = $arg->getAttribute('strict')) {
```

Comment:
@xabbuh setting a reference as not strict is deprecated in 2.8 already. The option is a no-op in 3.0.
So the way to make a file work in 2.8, 3.x and 4.0 is to skip the attribute entirely and avoiding the usage of non-strict references (which are a source of bugs anyway, as it means you bypass the scope restrictions)